domain/externalcontroller/state/state.go

```diff
@@ -168,51 +166,79 @@ func (st *State) ImportExternalControllers(ctx context.Context, infos []crossmod
 
 func (st *State) updateExternalControllerTx(
 	ctx context.Context,
-	tx *sql.Tx,
+	tx *sqlair.TX,
 	ci crossmodel.ControllerInfo,
 ) error {
 	cID := ci.ControllerTag.Id()
-	q := `
+	updateControllerStmt, err := sqlair.Prepare(`
```

Comment:
Note overly fond of doing this preparation inside the transaction, but I can see it's done for re-use across other methods.

I do prefer the prior way of assignment to a variable bore preparation, just because it clearly isolates the statement from other mechanics.

Will SQLair work with a type alias? I'd be tempted to use one here just to reduce the repeated long type names.

Reply:
Ideally we would prepare the queries in some sort of `Init` function and reuse them, this is how the use of Prepare was envisioned.

I've moved the query strings to separate declarations that come before the prepare. 
 
SQLair doesn't work with aliased types sadly, aliases are completely removed at runtime.